### ISSUE TITLE
apt-get update required AFTER adding ppa

### ISSUE THREAD

**Issue by TimNN** [ngineered/nginx-php-fpm#14]
Hi,

it seems that in commit 195e6999c2e4cb56af25e3c2db3d08818e8dfc26 you moved the `apt-get update` command *before* the `apt-add-repository` command, which makes sense to install the latest `software-properties-common`. However you still have to update *after* adding the ppa because otherwise the nginx shipped with ubuntu will be installed and not the one from the ppa.

**Comment by richarvey** [comment]
I'll get this fixed now, thanks for spotting that!

**Comment by richarvey** [comment]
New images should be building now.

**Comment by TimNN** [comment]
Awesome, thanks for the quick response!

**Comment by richarvey** [comment]
No worries would love to know your use case and if you need any more features.

**Comment by TimNN** [comment]
Actually I wanted *less* features, which is why I forked this repo and when testing my modifications noticed that on the 404 page it said `nginx/1.4.6 (Ubuntu)` which kinda confused me :D.

I needed an image which I could use to build other images on top on, such as a piwik image (there are existing piwik images, however neither was quite right for my use case).

You can find my changes at: https://github.com/TimNN/nginx-php-fpm/commit/7b3e7ec35fd5f6ebde4dd359ddfd10779374eefa

**Comment by richarvey** [comment]
Would it be useful if i build a base image (just nginx and php) and then a new image from that base on top that has the git features etc etc?

**Comment by TimNN** [comment]
Sure, that would be cool.

I made some other changes as you might have noticed which I'll just list here along with the reason for the change:

* I removed the increased upload / body sizes, because I don't think I'll ever need them and I'd rather change these on a case by case basis.
* I removed the `VOLUME` because I don't need it for my use case and again rather enable it on a case by case basis.
* I created a separate directory for my page sources (`www-root`) so they don't conflict with existing nginx files (I still use `/usr/share/nginx/html` for nginx error pages).
* I use an `ONBUILD` instruction to clear the root directory when building an image based on this image so that the phpinfo in the default index.php is not accidentally accessible.
* I reenabled the 404 error page because I saw no reason to disable it.
* I removed the `sendfile off` because while it might be required when running docker in virtual box, I usually don't do so, so this would needlessly impact performance.
